src/colorizer/ColorizeCanvas.ts

```diff
@@ -185,8 +287,65 @@ export default class ColorizeCanvas {
     this.setUniform("outlierColor", color);
   }
 
-  setHighlightedId(id: number): void {
-    this.setUniform("highlightedId", id);
+  setSelectedTrack(track: Track | null): void {
+    if (this.track && this.track?.trackId === track?.trackId) {
+      return;
+    }
+    this.track = track;
+    if (!track || !track.centroids || track.centroids.length === 0 || !this.frameResolution) {
+      return;
+    }
+    // Make a new array of the centroid positions in pixel coordinates.
+    // Points are in 3D while centroids are pairs of 2D coordinates in a 1D array
+    this.points = new Float32Array(track.length() * 3);
+    for (let i = 0; i < track.length(); i++) {
+      // Points passed directly as pixel coordinates relative to frame resolution.
+      this.points[3 * i + 0] = track.centroids[2 * i];
+      this.points[3 * i + 1] = track.centroids[2 * i + 1];
+      this.points[3 * i + 2] = 1;
+    }
+    // Assign new BufferAttribute because the old array has been discarded.
+    this.line.geometry.setAttribute("position", new BufferAttribute(this.points, 3));
+    this.line.geometry.getAttribute("position").needsUpdate = true;
+    this.updateTrackRange();
+  }
+
+  setShowTrackPath(show: boolean): void {
+    this.showTrackPath = show;
+  }
+
+  /**
+   * Updates the range of the track path line so that it shows up the path up to the current
+   * frame.
+   */
+  updateTrackRange(): void {
+    // Do nothing if track doesn't exist or doesn't have centroid data
+    if (!this.track || !this.track.centroids) {
+      return;
+    }
+    if (!this.showTrackPath) {
+      // Hide path
+      this.line.geometry.setDrawRange(0, 0);
+      return;
+    }
+    const trackFirstFrame = this.track.times[0];
+    // Clamp path to track length (don't draw non-existent vertices)
+    const range = Math.min(this.currentFrame - trackFirstFrame, this.track.length());
+    this.line.geometry.setDrawRange(0, range);
+  }
+
+  private updateHighlightedId(): void {
+    // Get highlighted id
+    if (this.track) {
+      let id;
+      // Tracks of length 1 should not be offset by 1
+      if (this.track.length() === 1) {
+        id = this.track.getIdAtTime(this.currentFrame);
+      } else {
+        id = this.track.getIdAtTime(this.currentFrame) - 1;
+      }
+      this.setUniform("highlightedId", id);
```

Comment:
Might prefer:

1. `if (!this.track) { return; }` over `if (this.track) { ...continue }`
2. Two `setUniform` calls in both "if" branches taking a const value rather than one call and a declared undefined value

...but these are minor style things.